<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Methods in Macroecology and Macroevolution</title>
  <meta name="description" content="Practical notes for macro module">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Methods in Macroecology and Macroevolution" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Practical notes for macro module" />
  <meta name="github-repo" content="nhcooper123/macro-methods-module/MacroModule" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Methods in Macroecology and Macroevolution" />
  
  <meta name="twitter:description" content="Practical notes for macro module" />
  

<meta name="author" content="Natalie Cooper (natalie.cooper@nhm.ac.uk)">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="what-you-need-to-be-able-to-do-in-r-before-you-start.html">
<link rel="next" href="visualising-phylogenies-in-r.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Macro Methods Module</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Methods in Macroecology and Macroevolution</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#module-aims"><i class="fa fa-check"></i><b>1.1</b> Module Aims</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#datasets-and-practical-handouts"><i class="fa fa-check"></i><b>1.2</b> Datasets and practical handouts</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#how-to-use-these-materials"><i class="fa fa-check"></i><b>1.3</b> How to use these materials</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#module-outline"><i class="fa fa-check"></i><b>1.4</b> Module outline</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><i class="fa fa-check"></i><b>2</b> What you need to be able to do in R before you start</a><ul>
<li class="chapter" data-level="2.1" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>2.1</b> Installing R (and RStudio)</a></li>
<li class="chapter" data-level="2.2" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#setting-the-working-directory"><i class="fa fa-check"></i><b>2.2</b> Setting the working directory</a></li>
<li class="chapter" data-level="2.3" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#using-a-script"><i class="fa fa-check"></i><b>2.3</b> Using a script</a></li>
<li class="chapter" data-level="2.4" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#installing-and-loading-extra-packages-in-r"><i class="fa fa-check"></i><b>2.4</b> Installing and loading extra packages in R</a></li>
<li class="chapter" data-level="2.5" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#loading-and-viewing-your-data-in-r"><i class="fa fa-check"></i><b>2.5</b> Loading and viewing your data in R</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html"><i class="fa fa-check"></i><b>3</b> Diversity Indices in R</a><ul>
<li class="chapter" data-level="3.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#what-are-diversity-indices"><i class="fa fa-check"></i><b>3.1</b> What are diversity indices?</a><ul>
<li class="chapter" data-level="3.1.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#alpha-beta-and-gamma-diversity-whittaker-1960-1972"><i class="fa fa-check"></i><b>3.1.1</b> <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\gamma\)</span> diversity (Whittaker 1960, 1972)</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#practical-example-using-pokemon"><i class="fa fa-check"></i><b>3.2</b> Practical example using Pokemon…</a></li>
<li class="chapter" data-level="3.3" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#species-diversity-indices"><i class="fa fa-check"></i><b>3.3</b> Species diversity indices</a><ul>
<li class="chapter" data-level="3.3.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#alpha-diversity"><i class="fa fa-check"></i><b>3.3.1</b> <span class="math inline">\(\alpha\)</span> diversity</a></li>
<li class="chapter" data-level="3.3.2" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#beta-diversity"><i class="fa fa-check"></i><b>3.3.2</b> <span class="math inline">\(\beta\)</span> diversity</a></li>
<li class="chapter" data-level="3.3.3" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#gamma-diversity"><i class="fa fa-check"></i><b>3.3.3</b> <span class="math inline">\(\gamma\)</span> diversity</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#species-accumulation-curves-colwell-coddington-1994"><i class="fa fa-check"></i><b>3.4</b> Species accumulation curves (Colwell &amp; Coddington 1994)</a><ul>
<li class="chapter" data-level="3.4.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#practical-example"><i class="fa fa-check"></i><b>3.4.1</b> Practical example</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#references"><i class="fa fa-check"></i><b>3.5</b> References</a><ul>
<li class="chapter" data-level="3.5.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#extra-reading"><i class="fa fa-check"></i><b>3.5.1</b> Extra reading</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html"><i class="fa fa-check"></i><b>4</b> Visualising phylogenies in R</a><ul>
<li class="chapter" data-level="4.1" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#loading-your-phylogeny-and-data-into-r"><i class="fa fa-check"></i><b>4.1</b> Loading your phylogeny and data into R</a><ul>
<li class="chapter" data-level="4.1.1" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#reading-in-a-phylogeny-from-a-file"><i class="fa fa-check"></i><b>4.1.1</b> Reading in a phylogeny from a file</a></li>
<li class="chapter" data-level="4.1.2" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#reading-in-a-phylogeny-that-is-already-built-into-r"><i class="fa fa-check"></i><b>4.1.2</b> Reading in a phylogeny that is already built into R</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#basic-tree-viewing-in-r"><i class="fa fa-check"></i><b>4.2</b> Basic tree viewing in R</a></li>
<li class="chapter" data-level="4.3" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#adding-trait-data-to-trees-in-r"><i class="fa fa-check"></i><b>4.3</b> Adding trait data to trees in R</a><ul>
<li class="chapter" data-level="4.3.1" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#ancestral-state-reconstructions-on-continuous-data"><i class="fa fa-check"></i><b>4.3.1</b> Ancestral state reconstructions on continuous data</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#phylomorphospace-plots-in-r"><i class="fa fa-check"></i><b>4.4</b> Phylomorphospace plots in R</a></li>
<li class="chapter" data-level="4.5" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#phylogenies-and-maps"><i class="fa fa-check"></i><b>4.5</b> Phylogenies and maps!</a></li>
<li class="chapter" data-level="4.6" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#a-new-package-for-visualising-trees-ggtree"><i class="fa fa-check"></i><b>4.6</b> A new package for visualising trees: ggtree</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Generalised Least Squares (PGLS) in R</a><ul>
<li class="chapter" data-level="5.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#preparing-for-the-analysis"><i class="fa fa-check"></i><b>5.1</b> Preparing for the analysis</a><ul>
<li class="chapter" data-level="5.1.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#load-the-required-packages"><i class="fa fa-check"></i><b>5.1.1</b> Load the required packages</a></li>
<li class="chapter" data-level="5.1.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#reading-and-checking-your-data-in-r"><i class="fa fa-check"></i><b>5.1.2</b> Reading and checking your data in R</a></li>
<li class="chapter" data-level="5.1.3" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#reading-and-checking-your-phylogeny-in-r"><i class="fa fa-check"></i><b>5.1.3</b> Reading and checking your phylogeny in R</a></li>
<li class="chapter" data-level="5.1.4" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#manipulating-your-data-and-phylogeny-in-r"><i class="fa fa-check"></i><b>5.1.4</b> Manipulating your data and phylogeny in R</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#ordinary-least-squares-ols-regression"><i class="fa fa-check"></i><b>5.2</b> Ordinary least squares (OLS) regression</a></li>
<li class="chapter" data-level="5.3" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#phylogenetic-generalized-least-squares-models-pgls"><i class="fa fa-check"></i><b>5.3</b> Phylogenetic generalized least squares models (PGLS)</a><ul>
<li class="chapter" data-level="5.3.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#likelihood-profiles-for-lambda-in-pgls-models"><i class="fa fa-check"></i><b>5.3.1</b> Likelihood profiles for <span class="math inline">\(\lambda\)</span> in PGLS models</a></li>
<li class="chapter" data-level="5.3.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#model-diagnostics-for-pgls-models"><i class="fa fa-check"></i><b>5.3.2</b> Model diagnostics for PGLS models</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#estimating-phylogenetic-signal-for-one-variable"><i class="fa fa-check"></i><b>5.4</b> Estimating phylogenetic signal for one variable</a><ul>
<li class="chapter" data-level="5.4.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#pagels-lambda-pagel-19971999-freckleton-et-al-2002"><i class="fa fa-check"></i><b>5.4.1</b> Pagel’s <span class="math inline">\(\lambda\)</span> (Pagel 1997/1999, Freckleton et al 2002)</a></li>
<li class="chapter" data-level="5.4.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#blombergs-k-blomberg-et-al-2003"><i class="fa fa-check"></i><b>5.4.2</b> Blomberg’s K (Blomberg et al 2003)</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#references-1"><i class="fa fa-check"></i><b>5.5</b> References</a><ul>
<li class="chapter" data-level="5.5.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#extra-reading-1"><i class="fa fa-check"></i><b>5.5.1</b> Extra Reading</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html"><i class="fa fa-check"></i><b>6</b> Macroevolutionary models in R: Part 1 - continuous traits</a><ul>
<li class="chapter" data-level="6.1" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#preparing-for-the-analysis-1"><i class="fa fa-check"></i><b>6.1</b> Preparing for the analysis</a><ul>
<li class="chapter" data-level="6.1.1" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#load-packages-read-in-the-data-and-the-tree"><i class="fa fa-check"></i><b>6.1.1</b> Load packages, read in the data and the tree</a></li>
<li class="chapter" data-level="6.1.2" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#modify-the-tree-and-data-so-they-can-be-used-in-the-analyses."><i class="fa fa-check"></i><b>6.1.2</b> Modify the tree and data so they can be used in the analyses.</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#models-of-continuous-trait-evolution"><i class="fa fa-check"></i><b>6.2</b> Models of continuous trait evolution</a><ul>
<li class="chapter" data-level="6.2.1" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#the-brownian-motion-bm-model"><i class="fa fa-check"></i><b>6.2.1</b> The Brownian motion (BM model)</a></li>
<li class="chapter" data-level="6.2.2" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#the-ornstein-uhlenbeck-ou-model"><i class="fa fa-check"></i><b>6.2.2</b> The Ornstein-Uhlenbeck (OU) model</a></li>
<li class="chapter" data-level="6.2.3" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#the-early-burst-eb-model"><i class="fa fa-check"></i><b>6.2.3</b> The Early Burst (EB) model</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#fitting-the-models-using-fitcontinuous"><i class="fa fa-check"></i><b>6.3</b> Fitting the models using <code>fitContinuous</code></a><ul>
<li class="chapter" data-level="6.3.1" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#comparing-models-of-evolution"><i class="fa fa-check"></i><b>6.3.1</b> Comparing models of evolution</a></li>
<li class="chapter" data-level="6.3.2" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#problems-with-convergence-and-bounds"><i class="fa fa-check"></i><b>6.3.2</b> Problems with convergence and bounds</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#references-2"><i class="fa fa-check"></i><b>6.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html"><i class="fa fa-check"></i><b>7</b> Macroevolutionary models in R: Part 2 - discrete traits</a><ul>
<li class="chapter" data-level="7.1" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#preparing-for-the-analysis-2"><i class="fa fa-check"></i><b>7.1</b> Preparing for the analysis</a><ul>
<li class="chapter" data-level="7.1.1" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#load-packages-read-in-the-data-and-the-tree-1"><i class="fa fa-check"></i><b>7.1.1</b> Load packages, read in the data and the tree</a></li>
<li class="chapter" data-level="7.1.2" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#modify-the-tree-and-data-so-they-can-be-used-in-the-analyses.-1"><i class="fa fa-check"></i><b>7.1.2</b> Modify the tree and data so they can be used in the analyses.</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#fitting-models-of-evolution-to-discrete-data-regime-dependent-evolution"><i class="fa fa-check"></i><b>7.2</b> Fitting models of evolution to discrete data (regime dependent evolution)</a></li>
<li class="chapter" data-level="7.3" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#models-of-discrete-trait-evolution"><i class="fa fa-check"></i><b>7.3</b> Models of discrete trait evolution</a><ul>
<li class="chapter" data-level="7.3.1" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#mk1-all-rates-equal-er"><i class="fa fa-check"></i><b>7.3.1</b> Mk1 – all rates equal (ER)</a></li>
<li class="chapter" data-level="7.3.2" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#mk-symmetric-rates-sym"><i class="fa fa-check"></i><b>7.3.2</b> Mk – symmetric rates (SYM)</a></li>
<li class="chapter" data-level="7.3.3" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#mk-all-rates-different-ard"><i class="fa fa-check"></i><b>7.3.3</b> Mk – All Rates Different (ARD)</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#fitting-the-models-using-fitdiscrete"><i class="fa fa-check"></i><b>7.4</b> Fitting the models using <code>fitDiscrete</code></a></li>
<li class="chapter" data-level="7.5" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#ancestral-state-reconstructions"><i class="fa fa-check"></i><b>7.5</b> Ancestral state reconstructions</a></li>
<li class="chapter" data-level="7.6" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#fitting-models-using-ouwie"><i class="fa fa-check"></i><b>7.6</b> Fitting models using <code>OUwie</code></a><ul>
<li class="chapter" data-level="7.6.1" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#multi-rate-brownian-motion-bms"><i class="fa fa-check"></i><b>7.6.1</b> Multi-rate Brownian motion (BMS)</a></li>
<li class="chapter" data-level="7.6.2" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#multi-peak-ou-models"><i class="fa fa-check"></i><b>7.6.2</b> Multi-peak OU models</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#references-3"><i class="fa fa-check"></i><b>7.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html"><i class="fa fa-check"></i><b>8</b> Geometric Morphometrics in R</a><ul>
<li class="chapter" data-level="8.1" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#a-quick-introduction-to-geometric-morphometrics"><i class="fa fa-check"></i><b>8.1</b> A quick introduction to geometric morphometrics</a><ul>
<li class="chapter" data-level="8.1.1" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#what-is-morphometrics"><i class="fa fa-check"></i><b>8.1.1</b> What is morphometrics?</a></li>
<li class="chapter" data-level="8.1.2" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#landmarks"><i class="fa fa-check"></i><b>8.1.2</b> Landmarks</a></li>
<li class="chapter" data-level="8.1.3" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#collecting-data-and-adding-landmarks"><i class="fa fa-check"></i><b>8.1.3</b> Collecting data and adding landmarks</a></li>
<li class="chapter" data-level="8.1.4" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#measurement-error"><i class="fa fa-check"></i><b>8.1.4</b> Measurement error</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#a-practical-example-using-r"><i class="fa fa-check"></i><b>8.2</b> A practical example using R</a></li>
<li class="chapter" data-level="8.3" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#gathering-the-data"><i class="fa fa-check"></i><b>8.3</b> Gathering the data</a></li>
<li class="chapter" data-level="8.4" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#digitising-adding-landmarks-and-curves"><i class="fa fa-check"></i><b>8.4</b> Digitising (adding landmarks and curves)</a></li>
<li class="chapter" data-level="8.5" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#reading-in-your-data-and-plotting-it"><i class="fa fa-check"></i><b>8.5</b> Reading in your data and plotting it</a></li>
<li class="chapter" data-level="8.6" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#generalised-procrustes-superimposition-gpa"><i class="fa fa-check"></i><b>8.6</b> Generalised Procrustes Superimposition (GPA)</a></li>
<li class="chapter" data-level="8.7" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#principal-components-and-plotting"><i class="fa fa-check"></i><b>8.7</b> Principal components and plotting</a></li>
<li class="chapter" data-level="8.8" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#statistical-analyses-of-geometric-morphometric-datasets"><i class="fa fa-check"></i><b>8.8</b> Statistical analyses of geometric morphometric datasets</a></li>
<li class="chapter" data-level="8.9" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#regression"><i class="fa fa-check"></i><b>8.9</b> Regression</a></li>
<li class="chapter" data-level="8.10" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#multivariate-regression"><i class="fa fa-check"></i><b>8.10</b> Multivariate regression</a></li>
<li class="chapter" data-level="8.11" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#anova"><i class="fa fa-check"></i><b>8.11</b> ANOVA</a></li>
<li class="chapter" data-level="8.12" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#manova"><i class="fa fa-check"></i><b>8.12</b> MANOVA</a></li>
<li class="chapter" data-level="8.13" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#resources-for-learning-geometric-morphometrics"><i class="fa fa-check"></i><b>8.13</b> Resources for learning geometric morphometrics</a></li>
<li class="chapter" data-level="8.14" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#references-4"><i class="fa fa-check"></i><b>8.14</b> References</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><i class="fa fa-check"></i><b>9</b> BAMM: Bayesian Analysis of Macroevolutionary Mixtures</a><ul>
<li class="chapter" data-level="9.1" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#a-quick-intro-to-bamm"><i class="fa fa-check"></i><b>9.1</b> A quick intro to BAMM</a></li>
<li class="chapter" data-level="9.2" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#a-brief-aside-how-do-i-run-bamm"><i class="fa fa-check"></i><b>9.2</b> A brief aside: how do I run BAMM?</a></li>
<li class="chapter" data-level="9.3" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#what-does-a-bamm-analysis-produce"><i class="fa fa-check"></i><b>9.3</b> What does a BAMM analysis produce?</a><ul>
<li class="chapter" data-level="9.3.1" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#distinct-shift-configurations-and-how-to-summarise-them"><i class="fa fa-check"></i><b>9.3.1</b> Distinct shift configurations and how to summarise them</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#a-practical-example-looking-at-bamm-results-using-bammtools"><i class="fa fa-check"></i><b>9.4</b> A practical example looking at BAMM results using <code>BAMMtools</code></a></li>
<li class="chapter" data-level="9.5" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#assessing-mcmc-convergence"><i class="fa fa-check"></i><b>9.5</b> Assessing MCMC convergence</a></li>
<li class="chapter" data-level="9.6" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#analysis-of-rate-shifts-in-marsupial-phylogeny"><i class="fa fa-check"></i><b>9.6</b> Analysis of rate shifts in marsupial phylogeny</a><ul>
<li class="chapter" data-level="9.6.1" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#how-many-rate-shifts"><i class="fa fa-check"></i><b>9.6.1</b> How many rate shifts?</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#mean-phylorate-plot"><i class="fa fa-check"></i><b>9.7</b> Mean phylorate plot</a></li>
<li class="chapter" data-level="9.8" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#best-single-distinct-shift-configuration"><i class="fa fa-check"></i><b>9.8</b> Best single distinct shift configuration</a></li>
<li class="chapter" data-level="9.9" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#credible-sets-of-shifts"><i class="fa fa-check"></i><b>9.9</b> Credible sets of shifts</a></li>
<li class="chapter" data-level="9.10" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#clade-specific-evolutionary-rates"><i class="fa fa-check"></i><b>9.10</b> Clade specific evolutionary rates</a></li>
<li class="chapter" data-level="9.11" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#rate-through-time-analysis"><i class="fa fa-check"></i><b>9.11</b> Rate-through-time analysis</a></li>
<li class="chapter" data-level="9.12" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#macroevolutionary-cohort-analysis"><i class="fa fa-check"></i><b>9.12</b> Macroevolutionary cohort analysis</a></li>
<li class="chapter" data-level="9.13" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#assumptions-and-issues-with-bamm"><i class="fa fa-check"></i><b>9.13</b> Assumptions and issues with BAMM</a></li>
<li class="chapter" data-level="9.14" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#references-5"><i class="fa fa-check"></i><b>9.14</b> References</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html"><i class="fa fa-check"></i><b>10</b> Critical thinking about methods and analyses</a><ul>
<li class="chapter" data-level="10.1" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#how-to-critically-evaluate-a-paper"><i class="fa fa-check"></i><b>10.1</b> How to critically evaluate a paper</a><ul>
<li class="chapter" data-level="10.1.1" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#logicinterpretation"><i class="fa fa-check"></i><b>10.1.1</b> Logic/interpretation</a></li>
<li class="chapter" data-level="10.1.2" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#data"><i class="fa fa-check"></i><b>10.1.2</b> Data</a></li>
<li class="chapter" data-level="10.1.3" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#methods"><i class="fa fa-check"></i><b>10.1.3</b> Methods</a></li>
<li class="chapter" data-level="10.1.4" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#moving-forwards"><i class="fa fa-check"></i><b>10.1.4</b> Moving forwards</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#references-6"><i class="fa fa-check"></i><b>10.2</b> References</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="practice-questions.html"><a href="practice-questions.html"><i class="fa fa-check"></i><b>11</b> Practice Questions</a><ul>
<li class="chapter" data-level="11.1" data-path="practice-questions.html"><a href="practice-questions.html#diversity-indices-in-r-1"><i class="fa fa-check"></i><b>11.1</b> Diversity indices in R</a></li>
<li class="chapter" data-level="11.2" data-path="practice-questions.html"><a href="practice-questions.html#pgls-in-r"><i class="fa fa-check"></i><b>11.2</b> PGLS in R</a></li>
<li class="chapter" data-level="11.3" data-path="practice-questions.html"><a href="practice-questions.html#macroevolutionary-models"><i class="fa fa-check"></i><b>11.3</b> Macroevolutionary models</a></li>
<li class="chapter" data-level="11.4" data-path="practice-questions.html"><a href="practice-questions.html#geometric-morphometrics"><i class="fa fa-check"></i><b>11.4</b> Geometric morphometrics</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/nhcooper123/TeachingMaterials/MSc_Museum/MacroModule" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Methods in Macroecology and Macroevolution</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="diversity-indices-in-r" class="section level1">
<h1><span class="header-section-number">Practical 3</span> Diversity Indices in R</h1>
<p>The aims of this practical are to learn how to use R to estimate diversity indices and species accumulation curves. You will need some of these functions to complete your paleoecology assessment.</p>
<p>We will be using some made up data about Pokemon sightings within the Museum.</p>
<p><strong>REMEMBER</strong></p>
<ul>
<li>Download all of the data for the practical into a folder somewhere on your computer.</li>
<li>Set your working directory to this folder.</li>
<li>Start a new script for this practical.</li>
</ul>
<p>You will also need to install the following packages:</p>
<ul>
<li><code>vegan</code></li>
<li><code>picante</code></li>
</ul>
<hr />
<div id="what-are-diversity-indices" class="section level2">
<h2><span class="header-section-number">3.1</span> What are diversity indices?</h2>
<p>A diversity index is a measure of the “diversity” of an area. <em>Diversity</em> could be measured in terms of the numbers of species (or higher taxonomic groupings like genera, families, phyla), or other metrics such as number of haplotypes if you’re interested in genetic diversity, or number of functional groups for studies of functional diversity.</p>
<p>Many diversity indices also account for how evenly spread these different types are. For example, they identify whether there are there five species with 10 individuals of each (even), or five species with one species with 96 individuals and four species with one individual each (uneven). These differences in evenness may be as important for ecosystem function as the number of species.</p>
<p>There are about a million different diversity indices (OK this is a slight exaggeration but there are a lot check out the <a href="https://cran.r-project.org/web/packages/vegan/vegan.pdf">vegan package</a> vignette and search for “diversity”), and different people prefer different measures for different questions. Amusingly the <code>vegan</code> helpfile for the function <code>diversify</code> states “these indices are all very closely related (Hill 1973), and there is no reason to despise one more than others (but if you are a graduate student, don’t drag me in, but obey your Professor’s orders)”.</p>
<div id="alpha-beta-and-gamma-diversity-whittaker-1960-1972" class="section level3">
<h3><span class="header-section-number">3.1.1</span> <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\gamma\)</span> diversity (Whittaker 1960, 1972)</h3>
<ul>
<li><span class="math inline">\(\alpha\)</span> diversity is the mean species diversity in sites or habitats at a local scale.</li>
<li><span class="math inline">\(\beta\)</span> diversity is turnover in <span class="math inline">\(\alpha\)</span> diversity among different sites.</li>
<li><span class="math inline">\(\gamma\)</span> gamma diversity is diversity at the landscape level.</li>
</ul>
<p>For example, if we count the species in Hyde Park and Green Park, we’d have a measure of <span class="math inline">\(\alpha\)</span> diversity for each. <span class="math inline">\(\beta\)</span> diversity would measure the difference between species found in Hyde Park and those found in Green Park. <span class="math inline">\(\gamma\)</span> diversity would be all the species found across London.</p>
<p><span class="math inline">\(\beta\)</span> diversity is the hardest to measure, and there are far more metrics for measuring this than any others.</p>
</div>
</div>
<div id="practical-example-using-pokemon" class="section level2">
<h2><span class="header-section-number">3.2</span> Practical example using Pokemon…</h2>
<p>We’re going to play around with some invented data on sampling sites within the Museum and the Pokemon we’ve managed to find there (don’t complain about my unlikely Pokemon combinations, they’re just made up data with Pokemon names rather than A, B, C etc!).</p>
<div class="figure">
<img src="pokemon.png" alt="Who doesn’t love a practical handout being interrupted by an enormous Pikachu?" />
<p class="caption">Who doesn’t love a practical handout being interrupted by an enormous Pikachu?</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load picante and vegan</span>
<span class="kw">library</span>(picante)
<span class="kw">library</span>(vegan)</code></pre></div>
<p>Next read in the data and take a look at it.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pokemon &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;pokemon_communities.csv&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Look at the data</span>
pokemon</code></pre></div>
<p>For the <code>vegan</code> functions to work you need a matrix where the columns are the species names, the rows are the communities, and the contents of the matrix are the numbers of each species found at each site (or presence absence data as 0s and 1s). We can use the <code>sample2matrix</code> function in <code>picante</code> to do this easily.</p>
<p>Note that this only works if your first variable is the name of the site, your second is abundance and your third is the names of the species present.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pokemon.matrix &lt;-<span class="st"> </span><span class="kw">sample2matrix</span>(pokemon)

<span class="co"># Look at the matrix</span>
pokemon.matrix</code></pre></div>
<pre><code>##        Bulbasaur Caterpie Charmander Pidgey Pikachu Psyduck Rattata
## site01         5        0          3      0       0       0       0
## site02         0        0          0      6       0       0       6
## site03         1        0          1      1       1       0      10
## site04         0        0          2      0       1       0       0
## site05         0        2          0      4       0       0       3
## site06         0        0          0      0       0       1       0
## site07         3        3          3      3       0       0       3
## site08         5        5          0      0       0       0       0
## site09         0        0          0      0       0       0       0
## site10         0        0          6      3       0       0       0
##        Spearow Squirtle Zubat
## site01       0        0     0
## site02       1        0     0
## site03       1        0     0
## site04       0        0     0
## site05       0        3     0
## site06       0        0     0
## site07       3        3     0
## site08       0        0     0
## site09       0       10     0
## site10       0        0     2</code></pre>
</div>
<div id="species-diversity-indices" class="section level2">
<h2><span class="header-section-number">3.3</span> Species diversity indices</h2>
<div id="alpha-diversity" class="section level3">
<h3><span class="header-section-number">3.3.1</span> <span class="math inline">\(\alpha\)</span> diversity</h3>
<p>The simplest measure of <span class="math inline">\(\alpha\)</span> diversity is just the number of species in each site. You can easily extract this as follows.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">specnumber</span>(pokemon.matrix)</code></pre></div>
<pre><code>## site01 site02 site03 site04 site05 site06 site07 site08 site09 site10 
##      2      3      6      2      4      1      7      2      1      3</code></pre>
<p>Simpson’s and Shannon’s diversity indices can be estimated using the function <code>diversity</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Simpson&#39;s index</span>
<span class="kw">diversity</span>(pokemon.matrix, <span class="dt">index =</span> <span class="st">&quot;simpson&quot;</span>)</code></pre></div>
<pre><code>##    site01    site02    site03    site04    site05    site06    site07 
## 0.4687500 0.5680473 0.5333333 0.4444444 0.7361111 0.0000000 0.8571429 
##    site08    site09    site10 
## 0.5000000 0.0000000 0.5950413</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Shannon&#39;s index</span>
<span class="kw">diversity</span>(pokemon.matrix, <span class="dt">index =</span> <span class="st">&quot;shannon&quot;</span>)</code></pre></div>
<pre><code>##    site01    site02    site03    site04    site05    site06    site07 
## 0.6615632 0.9110175 1.1729935 0.6365142 1.3579779 0.0000000 1.9459101 
##    site08    site09    site10 
## 0.6931472 0.0000000 0.9949236</code></pre>
</div>
<div id="beta-diversity" class="section level3">
<h3><span class="header-section-number">3.3.2</span> <span class="math inline">\(\beta\)</span> diversity</h3>
<p>The function <code>betadiver</code> allows you to estimate all the <span class="math inline">\(\beta\)</span> diversity indices mentioned in Koleff et al. 2003. For help on which indices are included type:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">betadiver</span>(<span class="dt">help=</span><span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## 1 &quot;w&quot; = (b+c)/(2*a+b+c)
## 2 &quot;-1&quot; = (b+c)/(2*a+b+c)
## 3 &quot;c&quot; = (b+c)/2
## 4 &quot;wb&quot; = b+c
## 5 &quot;r&quot; = 2*b*c/((a+b+c)^2-2*b*c)
## 6 &quot;I&quot; = log(2*a+b+c) - 2*a*log(2)/(2*a+b+c) - ((a+b)*log(a+b) +
## (a+c)*log(a+c)) / (2*a+b+c)
## 7 &quot;e&quot; = exp(log(2*a+b+c) - 2*a*log(2)/(2*a+b+c) - ((a+b)*log(a+b)
## + (a+c)*log(a+c)) / (2*a+b+c))-1
## 8 &quot;t&quot; = (b+c)/(2*a+b+c)
## 9 &quot;me&quot; = (b+c)/(2*a+b+c)
## 10 &quot;j&quot; = a/(a+b+c)
## 11 &quot;sor&quot; = 2*a/(2*a+b+c)
## 12 &quot;m&quot; = (2*a+b+c)*(b+c)/(a+b+c)
## 13 &quot;-2&quot; = pmin(b,c)/(pmax(b,c)+a)
## 14 &quot;co&quot; = (a*c+a*b+2*b*c)/(2*(a+b)*(a+c))
## 15 &quot;cc&quot; = (b+c)/(a+b+c)
## 16 &quot;g&quot; = (b+c)/(a+b+c)
## 17 &quot;-3&quot; = pmin(b,c)/(a+b+c)
## 18 &quot;l&quot; = (b+c)/2
## 19 &quot;19&quot; = 2*(b*c+1)/(a+b+c)/(a+b+c-1)
## 20 &quot;hk&quot; = (b+c)/(2*a+b+c)
## 21 &quot;rlb&quot; = a/(a+c)
## 22 &quot;sim&quot; = pmin(b,c)/(pmin(b,c)+a)
## 23 &quot;gl&quot; = 2*abs(b-c)/(2*a+b+c)
## 24 &quot;z&quot; = (log(2)-log(2*a+b+c)+log(a+b+c))/log(2)</code></pre>
<p>Note that some of these are similarity indices, and some are dissimilarity indices. See Koleff et al. 2003 for more details. Two commonly used similarity indices are Jaccard’s index and Sorenson’s index which can be estimated as follows (note that completely different communities get a score of 0):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Jaccard&#39;s index</span>
<span class="kw">betadiver</span>(pokemon.matrix, <span class="dt">method =</span> <span class="st">&quot;j&quot;</span>)</code></pre></div>
<pre><code>##           site01    site02    site03    site04    site05    site06
## site02 0.0000000                                                  
## site03 0.3333333 0.5000000                                        
## site04 0.3333333 0.0000000 0.3333333                              
## site05 0.0000000 0.4000000 0.2500000 0.0000000                    
## site06 0.0000000 0.0000000 0.0000000 0.0000000 0.0000000          
## site07 0.2857143 0.4285714 0.6250000 0.1250000 0.5714286 0.0000000
## site08 0.3333333 0.0000000 0.1428571 0.0000000 0.2000000 0.0000000
## site09 0.0000000 0.0000000 0.0000000 0.0000000 0.2500000 0.0000000
## site10 0.2500000 0.2000000 0.2857143 0.2500000 0.1666667 0.0000000
##           site07    site08    site09
## site02                              
## site03                              
## site04                              
## site05                              
## site06                              
## site07                              
## site08 0.2857143                    
## site09 0.1428571 0.0000000          
## site10 0.2500000 0.0000000 0.0000000</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Shannon&#39;s index</span>
<span class="kw">betadiver</span>(pokemon.matrix, <span class="dt">method =</span> <span class="st">&quot;sor&quot;</span>)</code></pre></div>
<pre><code>##           site01    site02    site03    site04    site05    site06
## site02 0.0000000                                                  
## site03 0.5000000 0.6666667                                        
## site04 0.5000000 0.0000000 0.5000000                              
## site05 0.0000000 0.5714286 0.4000000 0.0000000                    
## site06 0.0000000 0.0000000 0.0000000 0.0000000 0.0000000          
## site07 0.4444444 0.6000000 0.7692308 0.2222222 0.7272727 0.0000000
## site08 0.5000000 0.0000000 0.2500000 0.0000000 0.3333333 0.0000000
## site09 0.0000000 0.0000000 0.0000000 0.0000000 0.4000000 0.0000000
## site10 0.4000000 0.3333333 0.4444444 0.4000000 0.2857143 0.0000000
##           site07    site08    site09
## site02                              
## site03                              
## site04                              
## site05                              
## site06                              
## site07                              
## site08 0.4444444                    
## site09 0.2500000 0.0000000          
## site10 0.4000000 0.0000000 0.0000000</code></pre>
<p>Note that the outputs here are pairwise matrices, as these indices measure the similarity among each pair of sites. You can estimate Whittaker’s original using <code>method = &quot;w&quot;</code> (this is a dissimilarity method so completely different communities get a score of 1).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">betadiver</span>(pokemon.matrix, <span class="dt">method =</span> <span class="st">&quot;w&quot;</span>)</code></pre></div>
<pre><code>##           site01    site02    site03    site04    site05    site06
## site02 1.0000000                                                  
## site03 0.5000000 0.3333333                                        
## site04 0.5000000 1.0000000 0.5000000                              
## site05 1.0000000 0.4285714 0.6000000 1.0000000                    
## site06 1.0000000 1.0000000 1.0000000 1.0000000 1.0000000          
## site07 0.5555556 0.4000000 0.2307692 0.7777778 0.2727273 1.0000000
## site08 0.5000000 1.0000000 0.7500000 1.0000000 0.6666667 1.0000000
## site09 1.0000000 1.0000000 1.0000000 1.0000000 0.6000000 1.0000000
## site10 0.6000000 0.6666667 0.5555556 0.6000000 0.7142857 1.0000000
##           site07    site08    site09
## site02                              
## site03                              
## site04                              
## site05                              
## site06                              
## site07                              
## site08 0.5555556                    
## site09 0.7500000 1.0000000          
## site10 0.6000000 1.0000000 1.0000000</code></pre>
</div>
<div id="gamma-diversity" class="section level3">
<h3><span class="header-section-number">3.3.3</span> <span class="math inline">\(\gamma\)</span> diversity</h3>
<p>In this example, <span class="math inline">\(\gamma\)</span> diversity is the total number of species found across all sites. We can very simply calculate this in R using the following code:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(<span class="kw">unique</span>(pokemon$Species))</code></pre></div>
<pre><code>## [1] 10</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># To view unique species</span>
<span class="kw">unique</span>(pokemon$Species)</code></pre></div>
<pre><code>##  [1] Bulbasaur  Charmander Pidgey     Rattata    Spearow    Pikachu   
##  [7] Squirtle   Caterpie   Psyduck    Zubat     
## 10 Levels: Bulbasaur Caterpie Charmander Pidgey Pikachu ... Zubat</code></pre>
<p>Of course in this example things are simple as we only included species we actually spotted at each site. If we had a more complex dataset, we might worry about species being listed in a site with an NA or a zero for the abundance value. In this case we can ask R to only add up species where abundance is not NA (<code>!is.na(pokemon$Abundance)</code>) and (<code>&amp;</code>) where it is not zero (<code>pokemon$Abundance != 0</code>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(<span class="kw">unique</span>(pokemon$Species[!<span class="kw">is.na</span>(pokemon$Abundance) &amp;<span class="st"> </span>pokemon$Abundance !=<span class="st"> </span><span class="dv">0</span>]))</code></pre></div>
<pre><code>## [1] 10</code></pre>
<p>If you want to test this out, we can remove the Zubat (which only occurs in site 10) and check.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Replace Zubat abundance with 0</span>
<span class="co"># [31,2] means the 31st row and 2nd (Abundance) column</span>
pokemon[<span class="dv">31</span>, <span class="dv">2</span>] &lt;-<span class="st"> </span><span class="dv">0</span>

<span class="kw">length</span>(<span class="kw">unique</span>(pokemon$Species[!<span class="kw">is.na</span>(pokemon$Abundance) &amp;<span class="st"> </span>pokemon$Abundance !=<span class="st"> </span><span class="dv">0</span>]))</code></pre></div>
<pre><code>## [1] 9</code></pre>
</div>
</div>
<div id="species-accumulation-curves-colwell-coddington-1994" class="section level2">
<h2><span class="header-section-number">3.4</span> Species accumulation curves (Colwell &amp; Coddington 1994)</h2>
<p>Often when we talk about species diversity we’re interested in the total diversity of an area or a site. For example, if we want to conserve a patch of woodland, we might need to know how many species in total live there. Sounds easy enough right? Just go out and sample the heck out of that woodland… The problem of course is that sampling is time consuming and expensive, and in conservation we don’t have much time or money. In addition, completely sampling all species in an area is difficult, especially for small, rare, shy species. Instead we often estimate species richness by various means. Species accumulation curves are one way to do this.</p>
<p>Species accumulation curves are graphs showing the <strong>cumulative number of species</strong> recorded in an area or site as a function of the <strong>cumulative sampling effort</strong> taken to search for them. Sampling effort can be number of quadrats, number of hours of sampling etc. for <span class="math inline">\(\alpha\)</span> diversity, or number of sites if trying to get an estimate of <span class="math inline">\(\gamma\)</span> diversity.</p>
<p>The idea is that as you sample more, you should get closer to discovering all the species in the area. The first few samples you take will probably have lots of species you haven’t recorded yet, but this rate of discovery should slow down. Eventually you hope that you’ll stop finding any new species to record so the curve will asymptote, but in reality sampling is rarely that thorough. Luckily we can use species accumulation curves to estimate where the curve would asymptote if we kept on sampling.</p>
<div id="practical-example" class="section level3">
<h3><span class="header-section-number">3.4.1</span> Practical example</h3>
<p>Let’s try this for our Pokemon, how many species might be hiding in the Museum if we sampled throughly?</p>
<p>We can use the <code>pokemon.matrix</code> we’ve already created and estimate the accumulation curve using the <code>vegan</code> function <code>specaccum</code>. There are lots of methods for estimating these curves but we will use <code>method = &quot;random&quot;</code>. This works as follows.</p>
<p>It randomly selects a site and calculates the initial richness, then randomly selects a second site and calculates the cumulative richness (i.e. the second site plus the first site), and repeats this until all the sites have been used. It then repeats this process 1000 times. You can change the number of times it does this using <code>permutations</code> but 1000 is about right - too few and the curves are not smooth, too many and it takes ages. The function outputs the mean cumulative richness and standard deviations for across all 1000 permutations. We do this because our curve will look different depending on which site we start with, and so will give a different total richness estimate. Randomising the order helps us get a better estimate of the total richness, and the standard error on that estimate.</p>
<p>To do this for our Pokemon:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pokemon.curve &lt;-<span class="st"> </span><span class="kw">specaccum</span>(pokemon.matrix, <span class="dt">method =</span> <span class="st">&quot;random&quot;</span>, <span class="dt">permutations =</span> <span class="dv">1000</span>)

<span class="co"># Look at the results</span>
pokemon.curve</code></pre></div>
<pre><code>## Species Accumulation Curve
## Accumulation method: random, with 1000 permutations
## Call: specaccum(comm = pokemon.matrix, method = &quot;random&quot;, permutations = 1000) 
## 
##                                                                        
## Sites    1.000000 2.000000 3.000000 4.000000 5.000000 6.000000 7.000000
## Richness 2.994000 5.200000 6.678000 7.727000 8.456000 8.947000 9.280000
## sd       1.831207 1.672961 1.303079 0.921586 0.743382 0.672788 0.619664
##                             
## Sites    8.000000 9.00000 10
## Richness 9.575000 9.80800 10
## sd       0.540988 0.39407  0</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Plot the curve</span>
<span class="kw">plot</span>(pokemon.curve, <span class="dt">ci.type =</span> <span class="st">&quot;poly&quot;</span>, <span class="dt">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="dt">ci.col =</span> <span class="st">&quot;lightblue&quot;</span>, 
     <span class="dt">lwd =</span> <span class="dv">2</span>, <span class="dt">ci.lty =</span> <span class="dv">0</span>, <span class="dt">xlab =</span> <span class="st">&quot;number of sites&quot;</span>, 
     <span class="dt">ylab =</span> <span class="st">&quot;cumulative number of Pokemon species&quot;</span>)</code></pre></div>
<p><img src="macro-module_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
<p><code>&quot;ci.type = &quot;poly&quot;</code> tells R that you want a shaded area showing the confidence intervals from your randomisations. You can play around with the colours etc. if you want to.</p>
<p>To demonstrate why we need the randomisations, look at two curves for just 1 permutation each.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pokemon.curve1 &lt;-<span class="st"> </span><span class="kw">specaccum</span>(pokemon.matrix, <span class="dt">method =</span> <span class="st">&quot;random&quot;</span>, <span class="dt">permutations =</span> <span class="dv">1</span>)
pokemon.curve2 &lt;-<span class="st"> </span><span class="kw">specaccum</span>(pokemon.matrix, <span class="dt">method =</span> <span class="st">&quot;random&quot;</span>, <span class="dt">permutations =</span> <span class="dv">1</span>)

<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))

<span class="kw">plot</span>(pokemon.curve1,  
    <span class="dt">xlab =</span> <span class="st">&quot;number of sites&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;cumulative number of Pokemon species&quot;</span>)

<span class="kw">plot</span>(pokemon.curve2, 
    <span class="dt">xlab =</span> <span class="st">&quot;number of sites&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;cumulative number of Pokemon species&quot;</span>)</code></pre></div>
<p><img src="macro-module_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</code></pre></div>
<p>Finally to estimate total species richness across all sites we can (again) use many different metrics. Some common ones include Chao 2 (Chao 1987), Jackknife and Bootstrapping approaches and these are easy to estimate using the <code>vegan</code> function <code>specpool</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">specpool</span>(pokemon.matrix)</code></pre></div>
<pre><code>##     Species chao  chao.se jack1 jack1.se    jack2     boot   boot.se  n
## All      10 11.8 3.394113  11.8 1.272792 12.68889 10.90352 0.7502033 10</code></pre>
<p>Estimates range from 10.9 <span class="math inline">\(\pm\)</span> 0.75 (bootstrap) to 11.8 <span class="math inline">\(\pm\)</span> 3.39 (Chao2). So we can be fairly confident there are between 10 and 15 Pokemon in the Museum.</p>
<hr />
</div>
</div>
<div id="references" class="section level2">
<h2><span class="header-section-number">3.5</span> References</h2>
<ul>
<li>Chao, A. (1987). Estimating the population size for capture-recapture data with unequal catchability. Biometrics 43, 783–791.</li>
<li>Colwell, R.K. &amp; Coddington, J.A. (1994). Estimating terrestrial biodiversity through extrapolation. Phil. Trans. Roy. Soc. London B 345, 101–118.</li>
<li>Hill, M. O. (1973) Diversity and evenness: a unifying notation and its consequences. Ecology, 54, 427–432</li>
<li>Koleff, P., Gaston, K.J. and Lennon, J.J. (2003) Measuring beta diversity for presence-absence data. Journal of Animal Ecology 72, 367–382.</li>
<li>Whittaker, R.H. (1960) Vegetation of Siskiyou mountains, Oregon and California. Ecological Monographs 30, 279–338.</li>
<li>Whittaker, R. H. (1972). Evolution and Measurement of Species Diversity. Taxon, 21, 213-251. <a href="doi:10.2307/1218190" class="uri">doi:10.2307/1218190</a></li>
</ul>
<div id="extra-reading" class="section level3">
<h3><span class="header-section-number">3.5.1</span> Extra reading</h3>
<ul>
<li>Whittaker, R. J. et al. (2001). Scale and species richness: towards a general, hierarchical theory of species diversity. Journal of Biogeography, 28, 453-470. <a href="doi:10.1046/j.1365-2699.2001.00563.x" class="uri">doi:10.1046/j.1365-2699.2001.00563.x</a></li>
<li>O’Hara, R.B. (2005). Species richness estimators: how many species can dance on the head of a pin? J. Anim. Ecol. 74, 375–386.</li>
</ul>
<div id="nice-series-of-papers-on-betadiversity" class="section level4">
<h4><span class="header-section-number">3.5.1.1</span> Nice series of papers on betadiversity</h4>
<ul>
<li>Tuomisto, H. (2010) A diversity of beta diversities: straightening up a concept gone awry. Part 1. Defining beta diversity as a function of alpha and gamma diversity. Ecography, 33, 2-22.</li>
<li>Tuomisto, H. (2010) A diversity of beta diversities: straightening up a concept gone awry. Part 2. Quantifying beta diversity and related phenomena. Ecography, 33, 23-45. <a href="doi:10.1111/j.1600-0587.2009.06148.x" class="uri">doi:10.1111/j.1600-0587.2009.06148.x</a></li>
<li>Tuomisto, H. 2010. A consistent terminology for quantifying species diversity? Yes, it does exist. Oecologia 4: 853–860. <a href="doi:10.1007/s00442-010-1812-0" class="uri">doi:10.1007/s00442-010-1812-0</a></li>
<li>Tuomisto, H. (2011) Commentary: do we have a consistent terminology for species diversity? Yes, if we choose to use it. Oecologia, 167, 903-911.</li>
</ul>
</div>
<div id="immensely-cool-new-approach-using-methods-developed-by-alan-turing" class="section level4">
<h4><span class="header-section-number">3.5.1.2</span> Immensely cool new approach using methods developed by Alan Turing</h4>
<ul>
<li>Chiu, C.H., Wang, Y.T., Walther, B.A. &amp; Chao, A. (2014). Improved nonparametric lower bound of species richness via a modified Good-Turing frequency formula. Biometrics 70, 671–682.</li>
</ul>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visualising-phylogenies-in-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/nhcooper123/TeachingMaterials/MSc_Museum/MacroModule/edit/master/02_DiversityIndicesinR.Rmd",
"text": "Edit"
},
"download": ["macro-module.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
