<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Methods in Macroecology and Macroevolution</title>
  <meta name="description" content="Practical notes for macro module">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Methods in Macroecology and Macroevolution" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Practical notes for macro module" />
  <meta name="github-repo" content="nhcooper123/macro-methods-module/MacroModule" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Methods in Macroecology and Macroevolution" />
  
  <meta name="twitter:description" content="Practical notes for macro module" />
  

<meta name="author" content="Natalie Cooper (natalie.cooper@nhm.ac.uk)">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="visualising-phylogenies-in-r.html">
<link rel="next" href="macroevolutionary-models-in-r-part-1-continuous-traits.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Macro Methods Module</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Methods in Macroecology and Macroevolution</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#module-aims"><i class="fa fa-check"></i><b>1.1</b> Module Aims</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#datasets-and-practical-handouts"><i class="fa fa-check"></i><b>1.2</b> Datasets and practical handouts</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#how-to-use-these-materials"><i class="fa fa-check"></i><b>1.3</b> How to use these materials</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#module-outline"><i class="fa fa-check"></i><b>1.4</b> Module outline</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><i class="fa fa-check"></i><b>2</b> What you need to be able to do in R before you start</a><ul>
<li class="chapter" data-level="2.1" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>2.1</b> Installing R (and RStudio)</a></li>
<li class="chapter" data-level="2.2" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#setting-the-working-directory"><i class="fa fa-check"></i><b>2.2</b> Setting the working directory</a></li>
<li class="chapter" data-level="2.3" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#using-a-script"><i class="fa fa-check"></i><b>2.3</b> Using a script</a></li>
<li class="chapter" data-level="2.4" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#installing-and-loading-extra-packages-in-r"><i class="fa fa-check"></i><b>2.4</b> Installing and loading extra packages in R</a></li>
<li class="chapter" data-level="2.5" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#loading-and-viewing-your-data-in-r"><i class="fa fa-check"></i><b>2.5</b> Loading and viewing your data in R</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html"><i class="fa fa-check"></i><b>3</b> Diversity Indices in R</a><ul>
<li class="chapter" data-level="3.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#what-are-diversity-indices"><i class="fa fa-check"></i><b>3.1</b> What are diversity indices?</a><ul>
<li class="chapter" data-level="3.1.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#alpha-beta-and-gamma-diversity-whittaker-1960-1972"><i class="fa fa-check"></i><b>3.1.1</b> <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\gamma\)</span> diversity (Whittaker 1960, 1972)</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#practical-example-using-pokemon"><i class="fa fa-check"></i><b>3.2</b> Practical example using Pokemon…</a></li>
<li class="chapter" data-level="3.3" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#species-diversity-indices"><i class="fa fa-check"></i><b>3.3</b> Species diversity indices</a><ul>
<li class="chapter" data-level="3.3.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#alpha-diversity"><i class="fa fa-check"></i><b>3.3.1</b> <span class="math inline">\(\alpha\)</span> diversity</a></li>
<li class="chapter" data-level="3.3.2" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#beta-diversity"><i class="fa fa-check"></i><b>3.3.2</b> <span class="math inline">\(\beta\)</span> diversity</a></li>
<li class="chapter" data-level="3.3.3" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#gamma-diversity"><i class="fa fa-check"></i><b>3.3.3</b> <span class="math inline">\(\gamma\)</span> diversity</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#species-accumulation-curves-colwell-coddington-1994"><i class="fa fa-check"></i><b>3.4</b> Species accumulation curves (Colwell &amp; Coddington 1994)</a><ul>
<li class="chapter" data-level="3.4.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#practical-example"><i class="fa fa-check"></i><b>3.4.1</b> Practical example</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#references"><i class="fa fa-check"></i><b>3.5</b> References</a><ul>
<li class="chapter" data-level="3.5.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#extra-reading"><i class="fa fa-check"></i><b>3.5.1</b> Extra reading</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html"><i class="fa fa-check"></i><b>4</b> Visualising phylogenies in R</a><ul>
<li class="chapter" data-level="4.1" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#loading-your-phylogeny-and-data-into-r"><i class="fa fa-check"></i><b>4.1</b> Loading your phylogeny and data into R</a><ul>
<li class="chapter" data-level="4.1.1" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#reading-in-a-phylogeny-from-a-file"><i class="fa fa-check"></i><b>4.1.1</b> Reading in a phylogeny from a file</a></li>
<li class="chapter" data-level="4.1.2" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#reading-in-a-phylogeny-that-is-already-built-into-r"><i class="fa fa-check"></i><b>4.1.2</b> Reading in a phylogeny that is already built into R</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#basic-tree-viewing-in-r"><i class="fa fa-check"></i><b>4.2</b> Basic tree viewing in R</a></li>
<li class="chapter" data-level="4.3" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#adding-trait-data-to-trees-in-r"><i class="fa fa-check"></i><b>4.3</b> Adding trait data to trees in R</a><ul>
<li class="chapter" data-level="4.3.1" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#ancestral-state-reconstructions-on-continuous-data"><i class="fa fa-check"></i><b>4.3.1</b> Ancestral state reconstructions on continuous data</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#phylomorphospace-plots-in-r"><i class="fa fa-check"></i><b>4.4</b> Phylomorphospace plots in R</a></li>
<li class="chapter" data-level="4.5" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#phylogenies-and-maps"><i class="fa fa-check"></i><b>4.5</b> Phylogenies and maps!</a></li>
<li class="chapter" data-level="4.6" data-path="visualising-phylogenies-in-r.html"><a href="visualising-phylogenies-in-r.html#a-new-package-for-visualising-trees-ggtree"><i class="fa fa-check"></i><b>4.6</b> A new package for visualising trees: ggtree</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Generalised Least Squares (PGLS) in R</a><ul>
<li class="chapter" data-level="5.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#preparing-for-the-analysis"><i class="fa fa-check"></i><b>5.1</b> Preparing for the analysis</a><ul>
<li class="chapter" data-level="5.1.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#load-the-required-packages"><i class="fa fa-check"></i><b>5.1.1</b> Load the required packages</a></li>
<li class="chapter" data-level="5.1.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#reading-and-checking-your-data-in-r"><i class="fa fa-check"></i><b>5.1.2</b> Reading and checking your data in R</a></li>
<li class="chapter" data-level="5.1.3" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#reading-and-checking-your-phylogeny-in-r"><i class="fa fa-check"></i><b>5.1.3</b> Reading and checking your phylogeny in R</a></li>
<li class="chapter" data-level="5.1.4" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#manipulating-your-data-and-phylogeny-in-r"><i class="fa fa-check"></i><b>5.1.4</b> Manipulating your data and phylogeny in R</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#ordinary-least-squares-ols-regression"><i class="fa fa-check"></i><b>5.2</b> Ordinary least squares (OLS) regression</a></li>
<li class="chapter" data-level="5.3" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#phylogenetic-generalized-least-squares-models-pgls"><i class="fa fa-check"></i><b>5.3</b> Phylogenetic generalized least squares models (PGLS)</a><ul>
<li class="chapter" data-level="5.3.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#likelihood-profiles-for-lambda-in-pgls-models"><i class="fa fa-check"></i><b>5.3.1</b> Likelihood profiles for <span class="math inline">\(\lambda\)</span> in PGLS models</a></li>
<li class="chapter" data-level="5.3.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#model-diagnostics-for-pgls-models"><i class="fa fa-check"></i><b>5.3.2</b> Model diagnostics for PGLS models</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#estimating-phylogenetic-signal-for-one-variable"><i class="fa fa-check"></i><b>5.4</b> Estimating phylogenetic signal for one variable</a><ul>
<li class="chapter" data-level="5.4.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#pagels-lambda-pagel-19971999-freckleton-et-al-2002"><i class="fa fa-check"></i><b>5.4.1</b> Pagel’s <span class="math inline">\(\lambda\)</span> (Pagel 1997/1999, Freckleton et al 2002)</a></li>
<li class="chapter" data-level="5.4.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#blombergs-k-blomberg-et-al-2003"><i class="fa fa-check"></i><b>5.4.2</b> Blomberg’s K (Blomberg et al 2003)</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#references-1"><i class="fa fa-check"></i><b>5.5</b> References</a><ul>
<li class="chapter" data-level="5.5.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#extra-reading-1"><i class="fa fa-check"></i><b>5.5.1</b> Extra Reading</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html"><i class="fa fa-check"></i><b>6</b> Macroevolutionary models in R: Part 1 - continuous traits</a><ul>
<li class="chapter" data-level="6.1" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#preparing-for-the-analysis-1"><i class="fa fa-check"></i><b>6.1</b> Preparing for the analysis</a><ul>
<li class="chapter" data-level="6.1.1" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#load-packages-read-in-the-data-and-the-tree"><i class="fa fa-check"></i><b>6.1.1</b> Load packages, read in the data and the tree</a></li>
<li class="chapter" data-level="6.1.2" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#modify-the-tree-and-data-so-they-can-be-used-in-the-analyses."><i class="fa fa-check"></i><b>6.1.2</b> Modify the tree and data so they can be used in the analyses.</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#models-of-continuous-trait-evolution"><i class="fa fa-check"></i><b>6.2</b> Models of continuous trait evolution</a><ul>
<li class="chapter" data-level="6.2.1" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#the-brownian-motion-bm-model"><i class="fa fa-check"></i><b>6.2.1</b> The Brownian motion (BM model)</a></li>
<li class="chapter" data-level="6.2.2" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#the-ornstein-uhlenbeck-ou-model"><i class="fa fa-check"></i><b>6.2.2</b> The Ornstein-Uhlenbeck (OU) model</a></li>
<li class="chapter" data-level="6.2.3" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#the-early-burst-eb-model"><i class="fa fa-check"></i><b>6.2.3</b> The Early Burst (EB) model</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#fitting-the-models-using-fitcontinuous"><i class="fa fa-check"></i><b>6.3</b> Fitting the models using <code>fitContinuous</code></a><ul>
<li class="chapter" data-level="6.3.1" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#comparing-models-of-evolution"><i class="fa fa-check"></i><b>6.3.1</b> Comparing models of evolution</a></li>
<li class="chapter" data-level="6.3.2" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#problems-with-convergence-and-bounds"><i class="fa fa-check"></i><b>6.3.2</b> Problems with convergence and bounds</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="macroevolutionary-models-in-r-part-1-continuous-traits.html"><a href="macroevolutionary-models-in-r-part-1-continuous-traits.html#references-2"><i class="fa fa-check"></i><b>6.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html"><i class="fa fa-check"></i><b>7</b> Macroevolutionary models in R: Part 2 - discrete traits</a><ul>
<li class="chapter" data-level="7.1" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#preparing-for-the-analysis-2"><i class="fa fa-check"></i><b>7.1</b> Preparing for the analysis</a><ul>
<li class="chapter" data-level="7.1.1" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#load-packages-read-in-the-data-and-the-tree-1"><i class="fa fa-check"></i><b>7.1.1</b> Load packages, read in the data and the tree</a></li>
<li class="chapter" data-level="7.1.2" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#modify-the-tree-and-data-so-they-can-be-used-in-the-analyses.-1"><i class="fa fa-check"></i><b>7.1.2</b> Modify the tree and data so they can be used in the analyses.</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#fitting-models-of-evolution-to-discrete-data-regime-dependent-evolution"><i class="fa fa-check"></i><b>7.2</b> Fitting models of evolution to discrete data (regime dependent evolution)</a></li>
<li class="chapter" data-level="7.3" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#models-of-discrete-trait-evolution"><i class="fa fa-check"></i><b>7.3</b> Models of discrete trait evolution</a><ul>
<li class="chapter" data-level="7.3.1" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#mk1-all-rates-equal-er"><i class="fa fa-check"></i><b>7.3.1</b> Mk1 – all rates equal (ER)</a></li>
<li class="chapter" data-level="7.3.2" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#mk-symmetric-rates-sym"><i class="fa fa-check"></i><b>7.3.2</b> Mk – symmetric rates (SYM)</a></li>
<li class="chapter" data-level="7.3.3" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#mk-all-rates-different-ard"><i class="fa fa-check"></i><b>7.3.3</b> Mk – All Rates Different (ARD)</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#fitting-the-models-using-fitdiscrete"><i class="fa fa-check"></i><b>7.4</b> Fitting the models using <code>fitDiscrete</code></a></li>
<li class="chapter" data-level="7.5" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#ancestral-state-reconstructions"><i class="fa fa-check"></i><b>7.5</b> Ancestral state reconstructions</a></li>
<li class="chapter" data-level="7.6" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#fitting-models-using-ouwie"><i class="fa fa-check"></i><b>7.6</b> Fitting models using <code>OUwie</code></a><ul>
<li class="chapter" data-level="7.6.1" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#multi-rate-brownian-motion-bms"><i class="fa fa-check"></i><b>7.6.1</b> Multi-rate Brownian motion (BMS)</a></li>
<li class="chapter" data-level="7.6.2" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#multi-peak-ou-models"><i class="fa fa-check"></i><b>7.6.2</b> Multi-peak OU models</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="macroevolutionary-models-in-r-part-2-discrete-traits.html"><a href="macroevolutionary-models-in-r-part-2-discrete-traits.html#references-3"><i class="fa fa-check"></i><b>7.7</b> References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html"><i class="fa fa-check"></i><b>8</b> Geometric Morphometrics in R</a><ul>
<li class="chapter" data-level="8.1" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#a-quick-introduction-to-geometric-morphometrics"><i class="fa fa-check"></i><b>8.1</b> A quick introduction to geometric morphometrics</a><ul>
<li class="chapter" data-level="8.1.1" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#what-is-morphometrics"><i class="fa fa-check"></i><b>8.1.1</b> What is morphometrics?</a></li>
<li class="chapter" data-level="8.1.2" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#landmarks"><i class="fa fa-check"></i><b>8.1.2</b> Landmarks</a></li>
<li class="chapter" data-level="8.1.3" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#collecting-data-and-adding-landmarks"><i class="fa fa-check"></i><b>8.1.3</b> Collecting data and adding landmarks</a></li>
<li class="chapter" data-level="8.1.4" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#measurement-error"><i class="fa fa-check"></i><b>8.1.4</b> Measurement error</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#a-practical-example-using-r"><i class="fa fa-check"></i><b>8.2</b> A practical example using R</a></li>
<li class="chapter" data-level="8.3" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#gathering-the-data"><i class="fa fa-check"></i><b>8.3</b> Gathering the data</a></li>
<li class="chapter" data-level="8.4" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#digitising-adding-landmarks-and-curves"><i class="fa fa-check"></i><b>8.4</b> Digitising (adding landmarks and curves)</a></li>
<li class="chapter" data-level="8.5" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#reading-in-your-data-and-plotting-it"><i class="fa fa-check"></i><b>8.5</b> Reading in your data and plotting it</a></li>
<li class="chapter" data-level="8.6" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#generalised-procrustes-superimposition-gpa"><i class="fa fa-check"></i><b>8.6</b> Generalised Procrustes Superimposition (GPA)</a></li>
<li class="chapter" data-level="8.7" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#principal-components-and-plotting"><i class="fa fa-check"></i><b>8.7</b> Principal components and plotting</a></li>
<li class="chapter" data-level="8.8" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#statistical-analyses-of-geometric-morphometric-datasets"><i class="fa fa-check"></i><b>8.8</b> Statistical analyses of geometric morphometric datasets</a></li>
<li class="chapter" data-level="8.9" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#regression"><i class="fa fa-check"></i><b>8.9</b> Regression</a></li>
<li class="chapter" data-level="8.10" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#multivariate-regression"><i class="fa fa-check"></i><b>8.10</b> Multivariate regression</a></li>
<li class="chapter" data-level="8.11" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#anova"><i class="fa fa-check"></i><b>8.11</b> ANOVA</a></li>
<li class="chapter" data-level="8.12" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#manova"><i class="fa fa-check"></i><b>8.12</b> MANOVA</a></li>
<li class="chapter" data-level="8.13" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#resources-for-learning-geometric-morphometrics"><i class="fa fa-check"></i><b>8.13</b> Resources for learning geometric morphometrics</a></li>
<li class="chapter" data-level="8.14" data-path="geometric-morphometrics-in-r.html"><a href="geometric-morphometrics-in-r.html#references-4"><i class="fa fa-check"></i><b>8.14</b> References</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><i class="fa fa-check"></i><b>9</b> BAMM: Bayesian Analysis of Macroevolutionary Mixtures</a><ul>
<li class="chapter" data-level="9.1" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#a-quick-intro-to-bamm"><i class="fa fa-check"></i><b>9.1</b> A quick intro to BAMM</a></li>
<li class="chapter" data-level="9.2" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#a-brief-aside-how-do-i-run-bamm"><i class="fa fa-check"></i><b>9.2</b> A brief aside: how do I run BAMM?</a></li>
<li class="chapter" data-level="9.3" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#what-does-a-bamm-analysis-produce"><i class="fa fa-check"></i><b>9.3</b> What does a BAMM analysis produce?</a><ul>
<li class="chapter" data-level="9.3.1" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#distinct-shift-configurations-and-how-to-summarise-them"><i class="fa fa-check"></i><b>9.3.1</b> Distinct shift configurations and how to summarise them</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#a-practical-example-looking-at-bamm-results-using-bammtools"><i class="fa fa-check"></i><b>9.4</b> A practical example looking at BAMM results using <code>BAMMtools</code></a></li>
<li class="chapter" data-level="9.5" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#assessing-mcmc-convergence"><i class="fa fa-check"></i><b>9.5</b> Assessing MCMC convergence</a></li>
<li class="chapter" data-level="9.6" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#analysis-of-rate-shifts-in-marsupial-phylogeny"><i class="fa fa-check"></i><b>9.6</b> Analysis of rate shifts in marsupial phylogeny</a><ul>
<li class="chapter" data-level="9.6.1" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#how-many-rate-shifts"><i class="fa fa-check"></i><b>9.6.1</b> How many rate shifts?</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#mean-phylorate-plot"><i class="fa fa-check"></i><b>9.7</b> Mean phylorate plot</a></li>
<li class="chapter" data-level="9.8" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#best-single-distinct-shift-configuration"><i class="fa fa-check"></i><b>9.8</b> Best single distinct shift configuration</a></li>
<li class="chapter" data-level="9.9" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#credible-sets-of-shifts"><i class="fa fa-check"></i><b>9.9</b> Credible sets of shifts</a></li>
<li class="chapter" data-level="9.10" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#clade-specific-evolutionary-rates"><i class="fa fa-check"></i><b>9.10</b> Clade specific evolutionary rates</a></li>
<li class="chapter" data-level="9.11" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#rate-through-time-analysis"><i class="fa fa-check"></i><b>9.11</b> Rate-through-time analysis</a></li>
<li class="chapter" data-level="9.12" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#macroevolutionary-cohort-analysis"><i class="fa fa-check"></i><b>9.12</b> Macroevolutionary cohort analysis</a></li>
<li class="chapter" data-level="9.13" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#assumptions-and-issues-with-bamm"><i class="fa fa-check"></i><b>9.13</b> Assumptions and issues with BAMM</a></li>
<li class="chapter" data-level="9.14" data-path="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html"><a href="bamm-bayesian-analysis-of-macroevolutionary-mixtures.html#references-5"><i class="fa fa-check"></i><b>9.14</b> References</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html"><i class="fa fa-check"></i><b>10</b> Critical thinking about methods and analyses</a><ul>
<li class="chapter" data-level="10.1" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#how-to-critically-evaluate-a-paper"><i class="fa fa-check"></i><b>10.1</b> How to critically evaluate a paper</a><ul>
<li class="chapter" data-level="10.1.1" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#logicinterpretation"><i class="fa fa-check"></i><b>10.1.1</b> Logic/interpretation</a></li>
<li class="chapter" data-level="10.1.2" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#data"><i class="fa fa-check"></i><b>10.1.2</b> Data</a></li>
<li class="chapter" data-level="10.1.3" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#methods"><i class="fa fa-check"></i><b>10.1.3</b> Methods</a></li>
<li class="chapter" data-level="10.1.4" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#moving-forwards"><i class="fa fa-check"></i><b>10.1.4</b> Moving forwards</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#references-6"><i class="fa fa-check"></i><b>10.2</b> References</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="practice-questions.html"><a href="practice-questions.html"><i class="fa fa-check"></i><b>11</b> Practice Questions</a><ul>
<li class="chapter" data-level="11.1" data-path="practice-questions.html"><a href="practice-questions.html#diversity-indices-in-r-1"><i class="fa fa-check"></i><b>11.1</b> Diversity indices in R</a></li>
<li class="chapter" data-level="11.2" data-path="practice-questions.html"><a href="practice-questions.html#pgls-in-r"><i class="fa fa-check"></i><b>11.2</b> PGLS in R</a></li>
<li class="chapter" data-level="11.3" data-path="practice-questions.html"><a href="practice-questions.html#macroevolutionary-models"><i class="fa fa-check"></i><b>11.3</b> Macroevolutionary models</a></li>
<li class="chapter" data-level="11.4" data-path="practice-questions.html"><a href="practice-questions.html#geometric-morphometrics"><i class="fa fa-check"></i><b>11.4</b> Geometric morphometrics</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/nhcooper123/TeachingMaterials/MSc_Museum/MacroModule" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Methods in Macroecology and Macroevolution</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="phylogenetic-generalised-least-squares-pgls-in-r" class="section level1">
<h1><span class="header-section-number">Practical 5</span> Phylogenetic Generalised Least Squares (PGLS) in R</h1>
<p>The aims of this practical are to learn how to use R to perform Phylogenetic Generalised Least Squares (PGLS) analyses, and to estimate phylogenetic signal.</p>
<p>We will be using the evolution of primate life-history variables as an example. These data come from the PanTHERIA database (Jones et al. 2009) and 10kTrees (Arnold et al. 2010). Note that this is an old version of 10kTrees, so if you want to use it in your research please download the newest version.</p>
<p><strong>REMEMBER</strong></p>
<ul>
<li>Download all of the data for the practical into a folder somewhere on your computer.</li>
<li>Set your working directory to this folder.</li>
<li>Start a new script for this practical.</li>
</ul>
<p>You will also need to install the following packages:</p>
<ul>
<li><code>ape</code></li>
<li><code>geiger</code></li>
<li><code>picante</code></li>
<li><code>caper</code></li>
</ul>
<hr />
<div id="preparing-for-the-analysis" class="section level2">
<h2><span class="header-section-number">5.1</span> Preparing for the analysis</h2>
<div id="load-the-required-packages" class="section level3">
<h3><span class="header-section-number">5.1.1</span> Load the required packages</h3>
<p>To begin we need to load the packages for this practical.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ape)
<span class="kw">library</span>(geiger)
<span class="kw">library</span>(picante)
<span class="kw">library</span>(caper)</code></pre></div>
</div>
<div id="reading-and-checking-your-data-in-r" class="section level3">
<h3><span class="header-section-number">5.1.2</span> Reading and checking your data in R</h3>
<p>The data are in a comma-delimited text file called <code>Primatedata.csv</code>. Load these data as follows. I am assuming you have set your working directory.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">primatedata &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;Primatedata.csv&quot;</span>)</code></pre></div>
<p>Check everything loaded correctly:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(primatedata)</code></pre></div>
<pre><code>## &#39;data.frame&#39;:    77 obs. of  9 variables:
##  $ Order          : Factor w/ 1 level &quot;Primates&quot;: 1 1 1 1 1 1 1 1 1 1 ...
##  $ Family         : Factor w/ 15 levels &quot;Aotidae&quot;,&quot;Atelidae&quot;,..: 2 2 2 14 3 3 3 4 4 4 ...
##  $ Binomial       : Factor w/ 77 levels &quot;Alouatta palliata&quot;,..: 5 6 7 8 9 10 11 15 16 17 ...
##  $ AdultBodyMass_g: num  6692 7582 8697 958 558 ...
##  $ GestationLen_d : num  138 226 228 164 154 ...
##  $ HomeRange_km2  : num  2.28 0.73 1.36 0.02 0.32 0.02 0.00212 0.51 0.16 0.24 ...
##  $ MaxLongevity_m : num  336 328 454 304 215 ...
##  $ SocialGroupSize: num  14.5 42 20 2.95 6.85 ...
##  $ SocialStatus   : int  2 2 2 2 2 2 2 2 2 2 ...</code></pre>
<p>As you can see, the data contains the following variables: <code>Order</code>, <code>Family</code>, <code>Binomial</code>, <code>AdultBodyMass_g</code>, <code>GestationLen_d</code>, <code>HomeRange_km2</code>, <code>MaxLongevity_m</code>, and <code>SocialGroupSize</code>.</p>
</div>
<div id="reading-and-checking-your-phylogeny-in-r" class="section level3">
<h3><span class="header-section-number">5.1.3</span> Reading and checking your phylogeny in R</h3>
<p>To load a tree you need either the function <code>read.tree</code> or <code>read.nexus</code>. <code>read.tree</code> can deal with a number of different types of data (including DNA) whereas <code>read.nexus</code> reads NEXUS files. We will use a NEXUS file of the consensus tree from 10kTrees.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">primatetree &lt;-<span class="kw">read.nexus</span>(<span class="st">&quot;consensusTree_10kTrees_Version2.nex&quot;</span>)</code></pre></div>
<p>Let’s examine the tree by typing:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">primatetree</code></pre></div>
<pre><code>## 
## Phylogenetic tree with 226 tips and 221 internal nodes.
## 
## Tip labels:
##  Allenopithecus_nigroviridis, Cercopithecus_ascanius, Cercopithecus_cephus, Cercopithecus_cephus_cephus, Cercopithecus_cephus_ngottoensis, Cercopithecus_diana, ...
## 
## Rooted; includes branch lengths.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(primatetree)</code></pre></div>
<pre><code>## List of 4
##  $ edge       : int [1:446, 1:2] 227 228 229 230 231 232 233 234 234 235 ...
##  $ edge.length: num [1:446] 4.95 17.69 19.65 8.12 4.82 ...
##  $ Nnode      : int 221
##  $ tip.label  : chr [1:226] &quot;Allenopithecus_nigroviridis&quot; &quot;Cercopithecus_ascanius&quot; &quot;Cercopithecus_cephus&quot; &quot;Cercopithecus_cephus_cephus&quot; ...
##  - attr(*, &quot;class&quot;)= chr &quot;phylo&quot;
##  - attr(*, &quot;order&quot;)= chr &quot;cladewise&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># It&#39;s usually a good idea to quickly plot the tree too</span>
<span class="co"># Increase the margins so the plot looks better</span>
<span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))
<span class="kw">plot</span>(primatetree, <span class="dt">cex =</span> <span class="fl">0.2</span>, <span class="dt">typ =</span> <span class="st">&quot;fan&quot;</span>)</code></pre></div>
<p><img src="macro-module_files/figure-html/unnamed-chunk-78-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Put the margins back to something sensible for graphs</span>
<span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>))</code></pre></div>
<p><code>primatetree</code> is a fully resolved tree with branch lengths. There are 226 species and 221 internal nodes.</p>
<p>Most R functions require your tree to be dichotomous, i.e. to have no polytomies. To check whether your tree is dichotomous use <code>is.binary.tree</code>. If this is FALSE, use <code>multi2di</code> to make the tree dichotomous. This function works by randomly resolving polytomies with zero-length branches.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">is.binary.tree</span>(primatetree) <span class="co"># We want this to be TRUE</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">primatetree &lt;-<span class="st"> </span><span class="kw">multi2di</span>(primatetree)

<span class="co"># Check this is now TRUE</span>
<span class="kw">is.binary.tree</span>(primatetree)</code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>Most functions also require the tree to be rooted, i.e., to have one taxon designated as the outgroup. Our tree is rooted but if you wanted to change the root, or root an unrooted tree use <code>root</code>. Note that here I’ve just chosen a random species <em>Saimiri sciureus</em> to be the root.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">primatetree.reroot &lt;-<span class="st"> </span><span class="kw">root</span>(primatetree, <span class="st">&quot;Saimiri_sciureus&quot;</span>)
<span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))
<span class="kw">plot</span>(primatetree.reroot, <span class="dt">cex =</span> <span class="fl">0.2</span>)</code></pre></div>
<p><img src="macro-module_files/figure-html/unnamed-chunk-81-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>))</code></pre></div>
</div>
<div id="manipulating-your-data-and-phylogeny-in-r" class="section level3">
<h3><span class="header-section-number">5.1.4</span> Manipulating your data and phylogeny in R</h3>
<div id="species-names-with-spaces" class="section level4">
<h4><span class="header-section-number">5.1.4.1</span> Species names with spaces</h4>
<p>Species names in the tree cannot contain spaces so they are generally written as Genus_species (the gap between the genus name and species name replaced by an underscore <code>_</code>). If the species names in the data are written as Genus species with a space, then you will have to replace the spaces with <code>_</code> so that they match up with the species names in the tree. You can do this as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">primatedata$Binomial &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, primatedata$Binomial)</code></pre></div>
<p><code>gsub</code> means general substitution. It replaces any instance of the first item (here it’s a space) with the second item (_) but only in the variable you tell it to (<code>primatedata$Binomial</code>).</p>
</div>
<div id="mismatches-between-species-in-your-data-and-phylogeny" class="section level4">
<h4><span class="header-section-number">5.1.4.2</span> Mismatches between species in your data and phylogeny</h4>
<p>Often you will have data for species which are not in your phylogeny and/or species in your phylogeny which are not in your data. Some functions in R can deal with this, others will produce an error telling you the tree and data do not match (e.g. most <code>ape</code> functions). It’s useful to know how to deal with this so I have provided code below.</p>
<p>Note that the <code>caper</code> function <code>comparative.data</code> (see below) matches up species names in the tree and data for you before you run any analyses. All <code>geiger</code> functions match the tree and the data too. However, these functions are only as good as their inputs. If you have even slightly misspelled a species name in the tree or the data it will automatically be dropped from the analyses. It is therefore <strong>very important</strong> to check this before running an analysis.</p>
<p>First we can use the function <code>name.check</code> to find out which names do not match.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">check &lt;-<span class="st"> </span><span class="kw">name.check</span>(<span class="dt">phy =</span> primatetree, <span class="dt">data =</span> primatedata, <span class="dt">data.names =</span> primatedata$Binomial)</code></pre></div>
<p>You can look at <code>check</code> by printing it, I won’t do this here as it produces a lot of output but take a look on your computer.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">check</code></pre></div>
<p><code>check</code> has two parts, <code>tree_not_data</code> for species in the tree but not in the dataset, and <code>data_not_tree</code> for species in the dataset but not in the tree.</p>
<p>We can remove species missing from the tree easily using <code>drop.tip</code>. You need to list the species which you do <strong>not</strong> want to select and then drop them from the tree instead of selecting the species you do want.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">primatetree &lt;-<span class="st"> </span><span class="kw">drop.tip</span>(primatetree, check$tree_not_data)</code></pre></div>
<p>In this case we don’t have any species in the tree missing from the data, <code>data_not_tree</code> contains no species. However, if you do, to remove species from the data which are not in the tree you can use <code>match</code> and <code>subset</code> as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">matches &lt;-<span class="st"> </span><span class="kw">match</span>(primatedata$Binomial, check$data_not_tree, <span class="dt">nomatch =</span> <span class="dv">0</span>)
primatedata &lt;-<span class="st"> </span><span class="kw">subset</span>(primatedata, matches ==<span class="st"> </span><span class="dv">0</span>)</code></pre></div>
<p><code>==</code> means <em>equals</em>. So this line of code selects species which do appear in the <code>data_not_tree</code> list of species, i.e. their value from <code>matches</code> is 0.</p>
<p>Always check this has worked as expected by checking the data and the phylogeny. In the first instance you can just use <code>str</code> to make sure you have the expected number of species in each:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(primatedata)</code></pre></div>
<pre><code>## &#39;data.frame&#39;:    77 obs. of  9 variables:
##  $ Order          : Factor w/ 1 level &quot;Primates&quot;: 1 1 1 1 1 1 1 1 1 1 ...
##  $ Family         : Factor w/ 15 levels &quot;Aotidae&quot;,&quot;Atelidae&quot;,..: 2 2 2 14 3 3 3 4 4 4 ...
##  $ Binomial       : chr  &quot;Ateles_belzebuth&quot; &quot;Ateles_geoffroyi&quot; &quot;Ateles_paniscus&quot; &quot;Callicebus_moloch&quot; ...
##  $ AdultBodyMass_g: num  6692 7582 8697 958 558 ...
##  $ GestationLen_d : num  138 226 228 164 154 ...
##  $ HomeRange_km2  : num  2.28 0.73 1.36 0.02 0.32 0.02 0.00212 0.51 0.16 0.24 ...
##  $ MaxLongevity_m : num  336 328 454 304 215 ...
##  $ SocialGroupSize: num  14.5 42 20 2.95 6.85 ...
##  $ SocialStatus   : int  2 2 2 2 2 2 2 2 2 2 ...</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(primatetree)</code></pre></div>
<pre><code>## List of 4
##  $ edge       : int [1:152, 1:2] 78 79 80 81 82 83 84 85 86 87 ...
##  $ edge.length: num [1:152] 4.95 17.69 19.65 8.12 4.82 ...
##  $ Nnode      : int 76
##  $ tip.label  : chr [1:77] &quot;Cercopithecus_ascanius&quot; &quot;Cercopithecus_cephus&quot; &quot;Cercopithecus_mitis&quot; &quot;Cercopithecus_neglectus&quot; ...
##  - attr(*, &quot;class&quot;)= chr &quot;phylo&quot;
##  - attr(*, &quot;order&quot;)= chr &quot;cladewise&quot;</code></pre>
<hr />
</div>
</div>
</div>
<div id="ordinary-least-squares-ols-regression" class="section level2">
<h2><span class="header-section-number">5.2</span> Ordinary least squares (OLS) regression</h2>
<p>Before we use phylogenetic comparative methods, we will quickly do an ordinary least squares (OLS) regression. Let’s assume that we’re interested in the relationship between primate body mass and gestation length. First we should look at the data. Note that both variables are highly skewed suggesting that we might need to log transform them. In fact, by log transforming the data, we are making the differences multiplicative, not additive. This helps to deal with the problem that a 2g increase in body mass for an elephant, is quite different to a 2g increase in mass for a mouse, but a 2% increase in body mass for both is more comparable. Also note that the function <code>log</code> in R is actually natural log, not log10. And for fun let’s make everything rainbow colored!</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>)) <span class="co"># So we can see 4 plots at once</span>

<span class="co"># Raw data</span>
<span class="kw">hist</span>(primatedata$AdultBodyMass_g, <span class="dt">col =</span> <span class="kw">rainbow</span>(<span class="dv">8</span>), <span class="dt">main =</span> <span class="st">&quot;&quot;</span>)
<span class="kw">hist</span>(primatedata$GestationLen_d, <span class="dt">col =</span> <span class="kw">rainbow</span>(<span class="dv">8</span>), <span class="dt">main =</span> <span class="st">&quot;&quot;</span>)

<span class="co"># Log transformed</span>
<span class="kw">hist</span>(<span class="kw">log</span>(primatedata$AdultBodyMass_g), <span class="dt">col =</span> <span class="kw">rainbow</span>(<span class="dv">8</span>), <span class="dt">main =</span> <span class="st">&quot;&quot;</span>)
<span class="kw">hist</span>(<span class="kw">log</span>(primatedata$GestationLen_d), <span class="dt">col =</span> <span class="kw">rainbow</span>(<span class="dv">8</span>), <span class="dt">main =</span> <span class="st">&quot;&quot;</span>)</code></pre></div>
<p><img src="macro-module_files/figure-html/unnamed-chunk-89-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>)) <span class="co"># So we can go back to 1 plot</span></code></pre></div>
<p>OK let’s do a linear regression of log gestation length against log adult body mass:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">model.ols &lt;-<span class="st"> </span><span class="kw">lm</span>(<span class="kw">log</span>(GestationLen_d) ~<span class="st"> </span><span class="kw">log</span>(AdultBodyMass_g), <span class="dt">data =</span> primatedata)
<span class="kw">summary</span>(model.ols)</code></pre></div>
<pre><code>## 
## Call:
## lm(formula = log(GestationLen_d) ~ log(AdultBodyMass_g), data = primatedata)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -0.61614 -0.08279  0.00646  0.11414  0.50558 
## 
## Coefficients:
##                      Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)            4.1037     0.1108  37.042  &lt; 2e-16 ***
## log(AdultBodyMass_g)   0.1204     0.0141   8.544  1.1e-12 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.1977 on 75 degrees of freedom
## Multiple R-squared:  0.4933, Adjusted R-squared:  0.4865 
## F-statistic: 73.01 on 1 and 75 DF,  p-value: 1.097e-12</code></pre>
<p>The slope is positive (<code>0.1204 ± 0.0141</code>), and highly significant (note that R can’t display values lower than &lt;2e-16 which is why it shows up so often).</p>
<p>We can also plot the regression line on a scatter plot:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="kw">log</span>(GestationLen_d) ~<span class="st"> </span><span class="kw">log</span>(AdultBodyMass_g), <span class="dt">data =</span> primatedata)
<span class="kw">abline</span>(model.ols)</code></pre></div>
<p><img src="macro-module_files/figure-html/unnamed-chunk-91-1.png" width="672" /></p>
<p>We can look at the phylogenetic pseudoreplication problem on the graph by colouring the points by family.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="kw">log</span>(GestationLen_d) ~<span class="st"> </span><span class="kw">log</span>(AdultBodyMass_g), <span class="dt">data =</span> primatedata, 
    <span class="dt">col =</span> primatedata$Family, <span class="dt">pch =</span> <span class="dv">16</span>)</code></pre></div>
<p><img src="macro-module_files/figure-html/unnamed-chunk-92-1.png" width="672" /></p>
<hr />
</div>
<div id="phylogenetic-generalized-least-squares-models-pgls" class="section level2">
<h2><span class="header-section-number">5.3</span> Phylogenetic generalized least squares models (PGLS)</h2>
<p>There are several ways of accounting for phylogenetic non independence in your analyses. Here we will just cover one - phylogenetic generalized least squares (PGLS). Another popular earlier method is independent contrasts (PIC). This method is really similar to PGLS, in fact it is just a special kind of PGLS where <span class="math inline">\(\lambda\)</span> is equal to 1. PGLS offers some important advantages over independent contrasts. The model of trait evolution can be more flexible i.e., it can depart from a strict Brownian motion process (<span class="math inline">\(\lambda\)</span> or K = 1). Different scaling parameters (<span class="math inline">\(\lambda\)</span>, <span class="math inline">\(\kappa\)</span>, and <span class="math inline">\(\delta\)</span>) can be incorporated in the analysis, which can significantly improve the fit of the data to the model and thus also improve the estimation of the trait correlation. Another advantage of PGLS is that the intercept of the regression is not forced to be zero.</p>
<p>To perform PGLS models in R, <code>caper</code> requires you to first combine the phylogeny and data into one object using the function <code>comparative.data</code>.</p>
<p>Note that <code>vcv = TRUE</code> stores a variance covariance matrix of your tree (you will need this for the <code>pgls</code> function). <code>na.omit = FALSE</code> stops the function from removing species without data for all variables. <code>warn.dropped = TRUE</code> will tell you if any species are not in both the tree and the data and are therefore dropped from the comparative data object. Here we won’t drop any species because we already did this using <code>name.check</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">primate &lt;-<span class="st"> </span><span class="kw">comparative.data</span>(<span class="dt">phy =</span> primatetree, <span class="dt">data =</span> primatedata, 
                            <span class="dt">names.col =</span> Binomial, <span class="dt">vcv =</span> <span class="ot">TRUE</span>, 
                            <span class="dt">na.omit =</span> <span class="ot">FALSE</span>, <span class="dt">warn.dropped =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>If you do need to drop species, this function will give a warning telling you that some species have been dropped. You can view the dropped species using:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">primate$dropped$tips</code></pre></div>
<pre><code>## character(0)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">primate$dropped$unmatched.rows</code></pre></div>
<pre><code>## character(0)</code></pre>
<p><strong>Always</strong> make sure you check the list of dropped species is what you expected, it often reveals typos in your species names, or mismatches in taxonomies used etc. Here of course we already fixed this problem, so nothing is listed.</p>
<p>The function for PGLS analyses in <code>caper</code> is <code>pgls</code>. To fit a model which uses the Maximum Likelihood (ML) estimate of <span class="math inline">\(\lambda\)</span> we use the following code:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">model.pgls &lt;-<span class="st"> </span><span class="kw">pgls</span>(<span class="kw">log</span>(GestationLen_d) ~<span class="st"> </span><span class="kw">log</span>(AdultBodyMass_g), <span class="dt">data =</span> primate, 
                   <span class="dt">lambda =</span> <span class="st">&quot;ML&quot;</span>)
<span class="kw">summary</span>(model.pgls)</code></pre></div>
<pre><code>## 
## Call:
## pgls(formula = log(GestationLen_d) ~ log(AdultBodyMass_g), data = primate, 
##     lambda = &quot;ML&quot;)
## 
## Residuals:
##       Min        1Q    Median        3Q       Max 
## -0.098899 -0.011661  0.003082  0.017758  0.075133 
## 
## Branch length transformations:
## 
## kappa  [Fix]  : 1.000
## lambda [ ML]  : 0.892
##    lower bound : 0.000, p = 1.1435e-14
##    upper bound : 1.000, p = 0.00046393
##    95.0% CI   : (0.753, 0.967)
## delta  [Fix]  : 1.000
## 
## Coefficients:
##                      Estimate Std. Error t value  Pr(&gt;|t|)    
## (Intercept)          4.290229   0.160355 26.7546 &lt; 2.2e-16 ***
## log(AdultBodyMass_g) 0.104864   0.019628  5.3426 9.479e-07 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.0261 on 75 degrees of freedom
## Multiple R-squared: 0.2757,  Adjusted R-squared: 0.266 
## F-statistic: 28.54 on 1 and 75 DF,  p-value: 9.479e-07</code></pre>
<p>As well as the standard regression outputs, the output includes the estimated ML value of <span class="math inline">\(\lambda\)</span> (0.892) and p values from likelihood ratio tests showing whether the ML <span class="math inline">\(\lambda\)</span> is significantly different from 0 or 1. <span class="math inline">\(\kappa\)</span> and <span class="math inline">\(\delta\)</span> are also tree transformations which can improve the fit of the data to the tree. It is also possible to use <code>pgls</code> to optimise <span class="math inline">\(\kappa\)</span> or <span class="math inline">\(\delta\)</span> (using kappa = “ML” or delta = “ML” instead of lambda = “ML” in the code above). We will not cover this today. Note that optimizing more than one of these parameters at the same time is not advisable because it would be impossible to interpret the results!</p>
<p>We can also plot the results as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="kw">log</span>(GestationLen_d) ~<span class="st"> </span><span class="kw">log</span>(AdultBodyMass_g), <span class="dt">data =</span> primatedata)
<span class="kw">abline</span>(model.pgls)</code></pre></div>
<p><img src="macro-module_files/figure-html/unnamed-chunk-96-1.png" width="672" /></p>
<p>Sometimes you will find that <code>pgls</code> will not work and you get an <code>optim error</code>. This is much more common when using a Mac. To fix it all you need to do is change the bounds (upper and lower values) on the parameter being optimized, in this case <span class="math inline">\(\lambda\)</span>. Just change the lower bound of <span class="math inline">\(\lambda\)</span> to something a little bigger than 1e-6 until it works. For example:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">model.pgls2 &lt;-<span class="st"> </span><span class="kw">pgls</span>(<span class="kw">log</span>(GestationLen_d) ~<span class="st"> </span><span class="kw">log</span>(AdultBodyMass_g), <span class="dt">data =</span> primate,
                    <span class="dt">lambda =</span> <span class="st">&quot;ML&quot;</span>, <span class="dt">bounds =</span> <span class="kw">list</span>(<span class="dt">lambda =</span> <span class="kw">c</span>(<span class="fl">1e-05</span>, <span class="dv">1</span>)))</code></pre></div>
<div id="likelihood-profiles-for-lambda-in-pgls-models" class="section level3">
<h3><span class="header-section-number">5.3.1</span> Likelihood profiles for <span class="math inline">\(\lambda\)</span> in PGLS models</h3>
<p>You can look at the likelihood profiles for branch length transformations in PGLS models using <code>pgls.profile</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lambda.profile &lt;-<span class="st"> </span><span class="kw">pgls.profile</span>(model.pgls, <span class="st">&quot;lambda&quot;</span>)
<span class="kw">plot</span>(lambda.profile)</code></pre></div>
<p><img src="macro-module_files/figure-html/unnamed-chunk-98-1.png" width="672" /></p>
<p>This graph shows the likelihood profile of <span class="math inline">\(\lambda\)</span> in our model. Ideally you want a line with an obvious peak/optimum like this, rather than a flat line which would suggest <span class="math inline">\(\lambda\)</span> could be anything. You can see that the optimum (the peak of the curve) is at 0.892 as estimated in our PGLS model. The dotted red lines are the 95% confidence intervals on <span class="math inline">\(\lambda\)</span> for our model. <code>pgls.confint</code> prints out these numbers in <code>$ci.val</code></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pgls.confint</span>(model.pgls, <span class="st">&quot;lambda&quot;</span>)$ci.val</code></pre></div>
<pre><code>## [1] 0.753434 0.966543</code></pre>
<p><strong>Big problems with small datasets</strong> You will often find strange <span class="math inline">\(\lambda\)</span> profiles when you don’t have a lot of species in your data, because <span class="math inline">\(\lambda\)</span> (and Blomberg’s K - see below) has very low power to detect phylogenetic signal for less than 20-30 data points (see Freckleton et al. 2002 Am Nat). This means that using PGLS on small datasets is tricky - you almost always get ML <span class="math inline">\(\lambda\)</span> of zero but the <span class="math inline">\(\lambda\)</span> profile will show a pretty flat likelihood surface. Unfortunately people often forget to look at the <span class="math inline">\(\lambda\)</span> profile so erroneously conclude that there is no phylogenetic autocorrelation in their data.</p>
<p><strong>Generally I’d say don’t use small datasets, however, this seems unavoidable in some fields.</strong> Therefore my advice is to (only in this situation!) ignore one of Freckleton’s deadly sins (2009, JEB) and report the results from an OLS model (equivalent of PGLS with <span class="math inline">\(\lambda\)</span> = 0) and also report the results from a PGLS model with <span class="math inline">\(\lambda\)</span> set to 1 (equivalent to independent contrasts). This problem comes up every year and current consensus among the PCM community is that this is best solution at present, if collecting more data is really not an option!</p>
<p><strong>To set <span class="math inline">\(\lambda\)</span> to 1 you just replace “ML” with 1</strong></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">model.pgls3 &lt;-<span class="st"> </span><span class="kw">pgls</span>(<span class="kw">log</span>(GestationLen_d) ~<span class="st"> </span><span class="kw">log</span>(AdultBodyMass_g), 
                    <span class="dt">data =</span> primate, <span class="dt">lambda =</span> <span class="dv">1</span>)</code></pre></div>
</div>
<div id="model-diagnostics-for-pgls-models" class="section level3">
<h3><span class="header-section-number">5.3.2</span> Model diagnostics for PGLS models</h3>
<p>You should always check model diagnostic plots whenever you fit a model in R to check that your data meet the assumptions of the model. The method for this in PGLS is the same for OLS, independent contrasts and PGLS models (though the graphs are slightly different). To get model diagnostic plots for PGLS:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">0</span>))
<span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>))
<span class="kw">plot</span>(model.pgls)</code></pre></div>
<p><img src="macro-module_files/figure-html/unnamed-chunk-101-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</code></pre></div>
<p>Without going into the statistical details (and if you’ve no idea what these plots are for I suggest looking this up online or in a stats textbook), what you are looking for in these plots is:</p>
<ol style="list-style-type: decimal">
<li>In plot one you shouldn’t see any data with a studentized residual &gt; <span class="math inline">\(\pm\)</span> 3. Any species with such large residuals should be removed as these outliers may overly influence the results of the regression (see Jones and Purvis 1997). Often these are the result of measurement error associated with species pairs joined by very short branches. You should generally report results with and without outliers unless the results remain qualitatively the same.</li>
<li>The points of the Q-Q plot (plot 2) should approximately fall on the line. This tests for normality of residuals, one of the assumptions of linear models</li>
<li>Plots 3 and 4 should show a fairly random scattering of points. You want to avoid any clear patterns. The first is related to the systematic component of the model - any pattern here suggests that the model has not been correctly specified. The second is to test the assumption that variances are equal (homoscedascity).</li>
</ol>
<p>It takes practice to know what is “good”, “bad” and “acceptable” with these plots. I would say the plots above are fine, but there appear to be a couple of data points with studentized residuals &gt; <span class="math inline">\(\pm\)</span> 3 in plot 1 that should be removed, or at least checked for errors.</p>
<hr />
</div>
</div>
<div id="estimating-phylogenetic-signal-for-one-variable" class="section level2">
<h2><span class="header-section-number">5.4</span> Estimating phylogenetic signal for one variable</h2>
<div id="pagels-lambda-pagel-19971999-freckleton-et-al-2002" class="section level3">
<h3><span class="header-section-number">5.4.1</span> Pagel’s <span class="math inline">\(\lambda\)</span> (Pagel 1997/1999, Freckleton et al 2002)</h3>
<p>Phylogenetic signal is merely the pattern where close relatives have more similar trait values than more distant relatives (see Kamilar and Cooper 2013). Often people will mention that they “corrected for phylogeny” because of the phylogenetic signal in their variables. However, we <strong>do not</strong> correct for phylogeny because our variables show phylogenetic signal. We account for phylogenetic non-independence because the <strong>residuals</strong> from our models show phylogenetic signal (see Revell 2010). <span class="math inline">\(\lambda\)</span> shown in PGLS models above is the <span class="math inline">\(\lambda\)</span> for the model residuals <strong>not</strong> the individual variables.</p>
<p>Sometimes however, you might be interested in the phylogenetic signal of just one trait. <span class="math inline">\(\lambda\)</span> is really easy to estimate using <code>caper</code>. To do this for log GestationLen_d:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lambdaGL &lt;-<span class="st"> </span><span class="kw">pgls</span>(<span class="kw">log</span>(GestationLen_d) ~<span class="st"> </span><span class="dv">1</span>, <span class="dt">data =</span> primate, <span class="dt">lambda =</span> <span class="st">&quot;ML&quot;</span>)
<span class="kw">summary</span>(lambdaGL)</code></pre></div>
<pre><code>## 
## Call:
## pgls(formula = log(GestationLen_d) ~ 1, data = primate, lambda = &quot;ML&quot;)
## 
## Residuals:
##       Min        1Q    Median        3Q       Max 
## -0.128354 -0.014460  0.001389  0.017572  0.074329 
## 
## Branch length transformations:
## 
## kappa  [Fix]  : 1.000
## lambda [ ML]  : 0.948
##    lower bound : 0.000, p = &lt; 2.22e-16
##    upper bound : 1.000, p = 0.030039
##    95.0% CI   : (0.859, 0.996)
## delta  [Fix]  : 1.000
## 
## Coefficients:
##             Estimate Std. Error t value  Pr(&gt;|t|)    
## (Intercept)  5.00514    0.11723  42.694 &lt; 2.2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.0334 on 76 degrees of freedom
## Multiple R-squared:     0,   Adjusted R-squared:     0 
## F-statistic:   NaN on 0 and 76 DF,  p-value: NA</code></pre>
<p>Note that by replacing the explanatory variables with 1 you are just investigating the relationship between log gestation length and the phylogeny. Thus the <span class="math inline">\(\lambda\)</span> value is the <span class="math inline">\(\lambda\)</span> estimate for log GestationLen_d and the p values are from likelihood ratio tests showing whether the ML <span class="math inline">\(\lambda\)</span> is significantly different from 0 (no phylogenetic signal) or 1 (the expectation under Brownian motion).</p>
<hr />
</div>
<div id="blombergs-k-blomberg-et-al-2003" class="section level3">
<h3><span class="header-section-number">5.4.2</span> Blomberg’s K (Blomberg et al 2003)</h3>
<p>To estimate Blomberg’s K we use the <code>Kcalc</code> function in <code>picante</code>. First you need to set up a new vector with the values for the variable you are interested in, here the log of gestation length, and with a names attribute with the names of your species. Here I’ll call this vector <code>lngest</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lngest &lt;-<span class="st"> </span><span class="kw">log</span>(primatedata$GestationLen_d)
<span class="kw">names</span>(lngest) &lt;-<span class="st"> </span>primatedata$Binomial</code></pre></div>
<p>We can then calculate K for log gestation length:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">Kcalc</span>(lngest[primatetree$tip.label], primatetree)</code></pre></div>
<pre><code>##           [,1]
## [1,] 0.7757771</code></pre>
<p><code>Kcalc</code> (and <code>phylosignal</code>) require the trait data to be in the same order as the tree tip labels. <code>lngest[primatetree$tip.label]</code> selects gestation lengths from the species in the same order as they are in the tree (square brackets <code>[ ]</code> are used in R to subset data).</p>
<p>K for log gestation length is 0.7758. As with <span class="math inline">\(\lambda\)</span> above, we’re interested in whether the value of K is significantly different from what we’d expect by chance. We can do this using the function <code>phylosignal</code>. This function randomly assigns the trait values to the species and then calculates K. This is repeated 1000 times (if <code>reps = 1000</code>) and the observed value of K is then compared to the randomized values to determine its significance.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">phylosignal</span>(lngest[primatetree$tip.label], primatetree, <span class="dt">reps =</span> <span class="dv">1000</span>)</code></pre></div>
<pre><code>##           K PIC.variance.obs PIC.variance.rnd.mean PIC.variance.P
## 1 0.7757771      0.001660764            0.01212598    0.000999001
##   PIC.variance.Z
## 1      -2.292621</code></pre>
<p>In this case the observed K is significantly higher than the random values of K (<code>PIC.variance.P</code> &lt; 0.001). (Click on the little black arrows to scroll left along the table if you’re looking at this in RStudio).</p>
<p><em>Note this is a randomization test so your output here will not be identical to mine.</em></p>
<hr />
</div>
</div>
<div id="references-1" class="section level2">
<h2><span class="header-section-number">5.5</span> References</h2>
<ul>
<li>Arnold, C., L. J. Matthews, and C. L. Nunn. 2010. The 10ktrees website: a new online resource for primate phylogeny. Evolutionary Anthropology: Issues, News, and Reviews 19:114–118.</li>
<li>Blomberg, S. P., T. Garland, and A. R. Ives. 2003. Testing for phylogenetic signal in comparative data: behavioral traits are more labile. Evolution 57:717–745.</li>
<li>Freckleton, R. P., P. H. Harvey, and M. Pagel. 2002. Phylogenetic analysis and comparative data: a test and review of evidence. The American Naturalist 160:712-726.</li>
<li>Freckleton, R. P. (2009) The seven deadly sins of comparative analysis. Journal of Evolutionary Biology, 22, 1367–1375.</li>
<li>Jones, K. E., J. Bielby, M. Cardillo, S. A. Fritz, J. O’Dell, C. D. L. Orme, K. Safi, W. Sechrest, E. H. Boakes, C. Carbone, et al. 2009. Pantheria: a species-level database of life history, ecology, and geography of extant and recently extinct mammals: Ecological archives e090-184. Ecology 90:2648–2648.</li>
<li>Kamilar, J. M., &amp; Cooper, N. 2013. Phylogenetic signal in primate behaviour, ecology and life history. Phil. Trans. R. Soc. B, 368(1618), 20120341.</li>
<li>Pagel, M. 1999. Inferring the historical patterns of biological evolution. Nature, 401(6756), 877-884.</li>
<li>Revell, L. J. 2010. Phylogenetic signal and linear regression on species data. Methods in Ecology and Evolution, 1, 319-329.</li>
</ul>
<div id="extra-reading-1" class="section level3">
<h3><span class="header-section-number">5.5.1</span> Extra Reading</h3>
<ul>
<li>Losos, J.B. (2011) Seeing the forest for the trees: the limitations of phylogenies in comparative biology. The American Naturalist, 177, 709–727.</li>
</ul>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visualising-phylogenies-in-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="macroevolutionary-models-in-r-part-1-continuous-traits.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/nhcooper123/TeachingMaterials/MSc_Museum/MacroModule/edit/master/04_PGLSinR.Rmd",
"text": "Edit"
},
"download": ["macro-module.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
